Release/src/http/common/http_helpers.cpp

```diff
@@ -427,5 +433,308 @@ bool validate_method(const utility::string_t& method)
+            }
+
+            if (state() != Z_OK)
+            {
+                set_state(Z_STREAM_ERROR);
+                return data_buffer();
+            }
+
+            data_buffer output;
+            output.reserve(input_size);
+
+            const size_t BUFFER_SIZE = 1024;
+            uint8_t temp_buffer[BUFFER_SIZE];
+
+            size_t input_offset{ 0 };
+            auto flush = Z_NO_FLUSH;
+
+            while (flush == Z_NO_FLUSH)
+            {
+                auto total_out_before = stream().total_out;
+                auto available_input = input_size - input_offset;
+
+                if (available_input == 0)
+                {
+                    flush = finish ? Z_FINISH : Z_PARTIAL_FLUSH;
+                }
+                else
+                {
+                    stream().avail_in = static_cast<int>(available_input);
+                    stream().next_in = const_cast<uint8_t*>(&input[input_offset]);
+                }
+
+                do
+                {
+                    stream().next_out = temp_buffer;
+                    stream().avail_out = BUFFER_SIZE;
+
+                    set_state(deflate(&stream(), flush));
+
+                    if (has_error())
+                    {
+                        return data_buffer();
+                    }
+
+                    input_offset = read_output(input_offset, available_input, total_out_before, temp_buffer, output);
+
+                } while (stream().avail_out == 0);
+            }
+
+            return output;
+        }
+
+        ~stream_compressor_impl()
+        {
+            deflateEnd(&stream());
+        }
+    };
+#else // Stub impl for when compression is not supported
+
+    class compression_base_impl
+    {
+    public:
+        bool has_error() const
+        {
+            //Compression/Decompression has not been implemented or configured on this platform
+            if (!stream_compressor::is_supported())
+            {
+                throw std::exception();
+            } 
+            return true;
+        }
+    };
+
+    class stream_compressor::stream_compressor_impl : public compression_base_impl
+    {
+    public:
+        stream_compressor_impl(compression_algorithm) {}
+        compression::data_buffer compress(const uint8_t* data, size_t size, bool)
+        {
+            //Compression is not implemented or configured on this platform
+            if (!stream_compressor::is_supported())
+            {
+                throw std::exception();
+            }
+
+            return data_buffer(data, data + size);
+        }
+    };
+
+    class stream_decompressor::stream_decompressor_impl : public compression_base_impl
+    {
+    public:
+        stream_decompressor_impl(compression_algorithm) {}
+        compression::data_buffer decompress(const uint8_t* data, size_t size) 
+        {
+            //Decompression is not implemented or configured on this platform
+            if (!stream_decompressor::is_supported())
```

Comment:
Adding this conditional here seems to trick the LTCG into not seeing code that gets linked in after it as unreachable.